framework/src/userobjects/ElementSubdomainModifier.C

```diff
@@ -204,17 +223,161 @@ ElementSubdomainModifier::finalize()
 }
 
 void
-ElementSubdomainModifier::updateBoundaryInfo(MooseMesh & mesh,
-                                             const std::vector<const Elem *> & moved_elems)
+ElementSubdomainModifier::updateMovingBoundaryInfo(MooseMesh & mesh,
+                                                   const std::vector<const Elem *> & moved_elems)
 {
-  if (!_moving_boundary_specified && !_complement_moving_boundary_specified)
-    return;
+  if (_moving_boundary_subdomains.size())
+    mooseAssert(_moving_boundary_subdomains.size() == 2,
+                "The number of moving subdomains should be two");
+  /*
+    There are a couple of steps to reconstruct the moving boundary.
+    1) Retrieve all the active elements associated with the moving boundary
+     in a previous step. These elements and their neighbors will serve as
+     boundary element candidates for the current step.
+    2) Remove all the elements from the moving boundary. That literately
+     deletes the moving boundary from the database.
+    3) Append moved elements to the boundary element candidates.
+    4) Rectrouct the moving boundary using the boundary element candidates by
```

Comment:
```suggestion
    4) Reconstruct the moving boundary using the boundary element candidates by
```

Reply:
Reconstructing a boundary on sides shared by two subdomains is exactly what my separate moving boundary UO does...